Fix lecture edit page if active term is not set

fosterfarrell9 opened the pull request:
This fixes #682 by adding the missing case in the `older_than?` method in the `Lecture` model.

Splines reviewed (changes requested):
LGTM.

- Maybe you can add an end-to-end test to ensure that the lecture#edit page looks as expected? (instead of showing an error?)
- And maybe even add a unit test for the `lecture.stale?` method to ensure it gives proper results even in the case where we just have a lecture and not an active term set.

fosterfarrell9 commented:
I added unit tests for `lecture#stale?`, made `lecture#older_than?` private (it is not used anywhere except by the lecture#stale? method) and added a cypress test for the lecture edit page that should fail when the page throws an error.

Splines commented on spec/cypress/e2e/lecture_spec.cy.js:
```diff
@@ -0,0 +1,17 @@
+
+describe("Lecture edit page", () => {
+  it("shows content tab button", function () {
+    cy.createUserAndLogin("teacher").as("teacher");
+
+    cy.then(() => {
+      FactoryBot.create("lecture", "with_teacher_by_id",
+        { teacher_id: this.teacher.id }).as("lecture");
+    });
```
```suggestion
    cy.createUserAndLogin("teacher").then((teacher) => {
      FactoryBot.create("lecture", "with_teacher_by_id",
        { teacher_id: teacher.id }).as("lecture");
    });
```

Splines commented on spec/cypress/e2e/lecture_spec.cy.js:
```diff
@@ -0,0 +1,17 @@
+
+    cy.then(() => {
+      cy.visit(`/lectures/${this.lecture.id}/edit`);
+      cy.getBySelector("content-tab-btn").should("be.visible");
```
LGTM, this will definitely fail if the page throws an error ;)

Splines commented on app/models/lecture.rb:
```diff
@@ -830,12 +830,6 @@ def speakers
-    term.begin_date <= Term.active.begin_date - timespan
-  end
-
   def stale?
```
Maybe one last thing: could you add a docstring including an explanation for what "stale" means? (doesn't have to include every detail, just a high-level overview maybe)

Splines commented on spec/models/lecture_spec.rb:
```diff
@@ -86,6 +86,52 @@
+    end
+
+    context "when there is an active term" do
+      let(:year) { 2024 }
```
```suggestion
```

Splines commented on spec/models/lecture_spec.rb:
```diff
@@ -86,6 +86,52 @@
   end
 
+  describe "#stale?" do
+    let(:year) { 2024 }
```
```suggestion
```

Splines commented on spec/cypress/e2e/lecture_spec.cy.js:
rename to `lectures_spec.cy.js` (plural) to be consistent with other cypress spec files

fosterfarrell9 replied:
Thanks for catching this!

Splines reviewed (approved):
LGTM. Note that you'll have to approve the tests such that they can run through before merging via `Squash and merge`.

fosterfarrell9 commented:
run tests
